### DISCUSSION TITLE
clone API Call doesn't answer

### DISCUSSION THREAD

**Discussion by JoergBruenner** [DependencyTrack/dependency-track#2204] [Q&A]
Hi,

I'm using DT 4.6.3 as provided in a docker desktop environment. I'm tring to get the clone API call running. I can send the
API call, but there is no answer as expected. The work is done surprisingly. The logs telling me, the project was cloned. I can see the clone in the GUI. The API Key belongs to an administrator.
curl by Swagger:
curl -X PUT "http://localhost:8081/api/v1/project/clone" -H  "accept: application/json" -H  "X-Api-Key: Ga...m" -H  "Content-Type: application/json" -d "{\"project\":\"7c1a899b-2197-4cb9-839b-13b7ab1e85dc\",\"version\":\"2\"}"
Response Header:
 access-control-allow-credentials: true  access-control-allow-headers: Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count, *  access-control-allow-methods: GET, POST, PUT, DELETE, OPTIONS  access-control-allow-origin: *  access-control-expose-headers: Origin, Content-Type, Authorization, X-Requested-With, Content-Length, Accept, Origin, X-Api-Key, X-Total-Count  access-control-max-age: 3600  cache-control: private, max-age=0, must-revalidate, no-cache  content-length: 0  content-security-policy: default-src 'self';script-src 'self' 'unsafe-inline';style-src 'self' 'unsafe-inline';img-src 'self';connect-src 'self';font-src 'self';object-src 'self';media-src 'self';frame-src 'self';child-src 'self';form-action 'self';frame-ancestors 'none';  date: Tue, 29 Nov 2022 09:50:31 GMT  x-frame-options: DENY  x-powered-by: Dependency-Track v4.6.3 

What's wrong?

Thanks a lot for your help
Jörg

**Comment by nscuro** [comment]
Because cloning projects can be an expensive operation, the work is done asynchronously in the background. 

That way, there's no web server thread being blocked until the work is done.

**Comment by JoergBruenner** [comment]
Thank you Niklas.I'm trying to automate BoM handling. If I have to add a new version of an existing project I'd like to clone that project and load the new BoM to the clone up. With your hint it seems I have to use the search calls to find my cloned project, havn't I? Von meinem/meiner Galaxy gesendet
-------- Ursprüngliche Nachricht --------Von: Niklas ***@***.***> Datum: 29.11.22  15:09  (GMT+01:00) An: DependencyTrack/dependency-track ***@***.***> Cc: JoergBruenner ***@***.***>, Author ***@***.***> Betreff: Re: [DependencyTrack/dependency-track] clone API Call doesn't answer
  (Discussion #2204) 
Because cloning projects can be an expensive operation, the work is done asynchronously in the background.
That way, there's no web server thread being blocked until the work is done.
